src/mlpack/core/optimizers/cmaes/cmaes.hpp

```diff
@@ -0,0 +1,177 @@
+/**
+ * @file cmaes.hpp
+ * @author Kartik Nighania (GSoC 17 mentor Marcus Edel)
+ *
+ * Covariance Matrix Adaptation Evolution Strategy
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_CMAES_CMAES_HPP
+#define MLPACK_CORE_OPTIMIZERS_CMAES_CMAES_HPP
+
+#include <mlpack/core/math/random.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+class CMAES
+{
+ public:
+CMAES(int objectDim, arma::mat& start, arma::mat& stdDivs,
```

Comment:
We should add a comment for each function, and use the same style as for the rest of the codebase, so that doxygen can pick it up to generate some neat API documentation.